src/imageio/storage/piwigo.c

```diff
@@ -956,33 +1013,20 @@ int store(dt_imageio_module_storage_t *self, dt_imageio_module_data_t *sdata, co
   dt_storage_piwigo_gui_data_t *ui = self->gui_data;
 
   gint result = 0;
-
-  const char *ext = format->extension(fdata);
+  gint skipped = 0;
 
   // Let's upload image...
 
-  /* construct a temporary file name */
-  char fname[PATH_MAX] = { 0 };
-  dt_loc_get_tmp_dir(fname, sizeof(fname));
-  g_strlcat(fname, "/darktable.XXXXXX.", sizeof(fname));
-  g_strlcat(fname, ext, sizeof(fname));
-
   char *caption = NULL;
   char *description = NULL;
   char *author = NULL;
 
-  gint fd = g_mkstemp(fname);
-  if(fd == -1)
-  {
-    dt_control_log("failed to create temporary image for piwigo export");
-    fprintf(stderr, "failed to create tempfile: %s\n", fname);
-    return 1;
-  }
-  close(fd);
+  dt_image_t *img = dt_image_cache_get(darktable.image_cache, imgid, 'r');
```

Comment:
you've move the cache_get from below but now it is not always released (see `dt_image_cache_read_release`) and this is probably the cause of the lock on my side.